javagen/src/main/java/com/azure/autorest/template/ServiceClientBuilderTemplate.java

```diff
@@ -274,17 +277,17 @@ public final void write(ClientBuilder clientBuilder, JavaFile javaFile) {
                 if (settings.isDataPlaneClient()) {
                     serializerExpression = JACKSON_SERIALIZER;
                 } else {
-                    serializerExpression = getSerializerMemberName();
+                    serializerExpression = "build" + CodeNamer.toPascalCase(getSerializerMemberName());
                 }
 
                 if (settings.isFluent()) {
-                    function.line(String.format("%1$s client = new %2$s(pipeline, %3$s, defaultPollInterval, environment%4$s);",
+                    function.line(String.format("%1$s client = new %2$s(buildPipeline, %3$s, defaultPollInterval, environment%4$s);",
                             serviceClient.getClassName(),
                             serviceClient.getClassName(),
                             serializerExpression,
                             constructorArgs));
                 } else {
-                    function.line(String.format("%1$s client = new %2$s(pipeline, %3$s%4$s);",
+                    function.line(String.format("%1$s client = new %2$s(buildPipeline, %3$s%4$s);",
                             serviceClient.getClassName(), serviceClient.getClassName(), serializerExpression, constructorArgs));
```

Comment:
There is many `build` hardcoded, some is directly in string, some add to string.

Would it be better to use a method e.g. `getLocalBuildVariableName()`, and hide what the variable called within the method?

nit, personally I think `localPipeline` might be easier to understand? Or `myPipeline`, or `clientPipeline`.